h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -722,6 +724,9 @@ public ResultInterface queryMeta() {
 
     @Override
     protected ResultInterface queryWithoutCache(int maxRows, ResultTarget target) {
+        disableLazyForJoinSubqueries(topTableFilter.getJoin());
+        disableLazyForJoinSubqueries(topTableFilter.getNestedJoin());
```

Comment:
Perhaps there is nothing to do when lazy mode is not enabled (default) and these invocations need to be conditional? Or I'm missing something?

Reply:
Sure. Fixed.